apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/status/InheriteMatchExceptionCheckStrategy.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.context.status;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import org.apache.skywalking.apm.agent.core.boot.ServiceManager;
+
+public class InheriteMatchExceptionCheckStrategy implements ExceptionCheckStrategy {
+
+    private final HashSet<Class<? extends Throwable>> ignoredExceptions = new HashSet<>(16);
+
+    @Override
+    public boolean isError(final Throwable e) {
+        Class<? extends Throwable> clazz = e.getClass();
+        if (ignoredExceptions.contains(clazz)) {
+            return false;
+        }
+        StatusCheckService statusTriggerService = ServiceManager.INSTANCE.findService(StatusCheckService.class);
+        String[] ignoredExceptionNames = statusTriggerService.getIgnoredExceptionNames();
+        if (isInheritMatch(getSuperclasses(clazz), ignoredExceptionNames)) {
+            ignoredExceptions.add(clazz);
+            return false;
+        }
+        return true;
+    }
+
+    private boolean isInheritMatch(final List<String> superclasses, final String[] ignoredExceptionNames) {
+        for (final String superclass : superclasses) {
+            for (final String ignoreExceptionName : ignoredExceptionNames) {
+                if (superclass.equals(ignoreExceptionName)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    List<String> getSuperclasses(Class<? extends Throwable> clazz) {
+        List<String> superClasses = new ArrayList<>();
+        superClasses.add(clazz.getName());
+        Class<?> superclass = clazz.getSuperclass();
+        while (superclass != null && !superclass.equals(Exception.class)) {
+            superClasses.add(superclass.getName());
+            superclass = superclass.getSuperclass();
+        }
+        return superClasses;
+    }
```

Comment:
i know ，but i want to ignore classloader affect. we don't know how to load config class by which classloader.so I only judged it by class name.because of only little class in loop,it would not cost much time.

Reply:
If you want this, use this way
1. c = e.getClass().getClassLoader()
1. If c.forName(exception Class) has classNotFound, then should not ignore
1. If c.forName(exception Class) has a class reference, you could do `isAssignableFrom`